### DISCUSSION TITLE
For Open Thread dependency

### DISCUSSION THREAD

**Discussion by bchoineubility** [dartsim/dart#1782] [Q&A] [answered]
Hi, 
I built DART 6.9.0 with yocto, by the way, configure log contain a warning message.

```Could NOT find OpenSceneGraph nor OpenThreads -- we will skip dart-gui-osg.```
So, I am googling for Open Thread, it would be a complete and independent RTOS not a library using Linux.

But, I am not sure why Open Thread is needed in the DART because DART can be run on the RTOS like.

Can you please explain for it?

BR,
Mark

**Comment by olzhas** [comment, marked as answer]
Hello Mark,

To the best of my knowledge, OpenSceneGraph is a library utilized for rendering dart simulations. Within OSG, there is a smaller library known as OpenThreads, which serves as an abstraction layer for cross-platform threading. So OpenThreads library is not related to that RTOS with a similar name that you discovered.

Kind regards,
Olzhas

**Reply by bchoineubility** [reply]
Thanks for kindly reply.

BR,
Mark

**Comment by jslee02** [comment]
Sorry for the very late maintainer follow-up, and thanks to Olzhas for the earlier answer. That answer is correct: in the DART build warning, `OpenThreads` means the threading support library that comes with OpenSceneGraph. It is not the OpenThread RTOS.

For DART 6.9 and the current DART 6.x line, this dependency is only for the optional `dart-gui-osg` visualizer/examples. Core DART, headless simulation, and non-OSG builds do not need it. If CMake prints `Could NOT find OpenSceneGraph nor OpenThreads -- we will skip dart-gui-osg`, the build can still be valid unless you specifically need the OSG GUI components. In that case, install OpenSceneGraph/OpenThreads or disable that component explicitly with `DART_BUILD_GUI_OSG=OFF`.

For DART 7/main, the legacy OpenSceneGraph GUI path has been retired; maintained visualization work is on the newer GUI/Filament direction instead.
